### ISSUE TITLE
<a> Tag Helper - Associated VS2017 Warning

### ISSUE THREAD

**Issue by NTaylorMullen** [aspnet/Razor#1428]
_From @stap123 on June 12, 2017 9:13_

When using the <a> tag helper and a target attribute simultaneously VS2017 gives a warning saying "Attribute is allowed only when 'href' is present".  I guess it's pretty obvious why it's happening as it tells us but not sure it's the desired behaviour.

Example of tag:

```
<a class="list-group-item" asp-controller="Home" asp-action="CPDRedirect" asp-route-id="2" target="_blank">View all activities</a>
```

This is obviously not stopping anything working but thought you might like to know (and it's a little annoying to have warnings when there isn't anything wrong).

If this isn't the correct repo, can you let me know the correct one and I will raise there.

_Copied from original issue: aspnet/Mvc#6391_

**Comment by NTaylorMullen** [comment]
@stap123 thanks for filing the issue! You're correct in ignoring the warnings; however, we should definitely make this work on our end.

We should consider adding an `OutputAttributeHint` to TagHelper bound attributes. This would work much like `OutputElementHint` but for attributes.

**Comment by rynowak** [comment]
@NTaylorMullen - should we add this as part of the 2.1 release? It will be a while before we can ship tooling for it, but if this is compelling we should probably add it to the runtime so it get adoption by the time tooling lands.

**Comment by NTaylorMullen** [comment]
@rynowak sure why not 😉 The tooling feature for this wouldn't make it until 15.8 FYI

**Comment by rynowak** [comment]
Yeah, but we don't know if there will be another runtime release for 15.8, so this is the safest bet. IMO it's ok to add this now and then light up the tooling later.

**Comment by NTaylorMullen** [comment]
Another external customer report: https://stackoverflow.com/a/49437763/1574622
